eng/common/scripts/Verify-Readme.ps1

```diff
@@ -1,24 +1,102 @@
 # Wrapper Script for Readme Verification
 [CmdletBinding()]
 param (
-  [Parameter(Mandatory = $true)]
+  [Parameter(Mandatory = $false)]
   [string]$DocWardenVersion,
-  [Parameter(Mandatory = $true)]
-  [string]$ScanPath,
   [string]$RepoRoot,
+  [string]$ScanPaths,
   [Parameter(Mandatory = $true)]
   [string]$SettingsPath
 )
+. (Join-Path $PSScriptRoot common.ps1)
+$DefaultDocWardenVersion = "0.7.2"
+$script:FoundError = $false
+
+function Test-Readme-Files {
+  param(
+    [string]$SettingsPath,
+    [string]$ScanPath,
+    [string]$RepoRoot)
+
+      Write-Host "Scanning..."
+
+      if ($RepoRoot)
+      {
+        Write-Host "ward scan -d $ScanPath -u $RepoRoot -c $SettingsPath"
+        ward scan -d $ScanPath -u $RepoRoot -c $SettingsPath
+      }
+      else
+      {
+        Write-Host "ward scan -d $ScanPath -c $SettingsPath"
+        ward scan -d $ScanPath -c $SettingsPath
+      }
+      # ward scan is what returns the non-zero exit code on failure.
+      # Since it's being called from a function, that error needs to
+      # be propagated back so the script can exit appropriately
+      if ($LASTEXITCODE -ne 0) {
+        $script:FoundError = $true
+      }
+}
 
+# Verify all of the inputs before running anything
+if ([String]::IsNullOrWhiteSpace($DocWardenVersion)) {
+  $DocWardenVersion = $DefaultDocWardenVersion
+}
+
+# verify the doc settings file exists
+if (!(Test-Path -Path $SettingsPath -PathType leaf)) {
+  LogError "Setting file, $SettingsPath, does not exist"
+  $script:FoundError = $true
+}
+
+# Verify that either ScanPath or ScanPaths were set but not both or neither
+if ([String]::IsNullOrWhiteSpace($ScanPaths)) {
+  LogError "ScanPaths cannot be empty."
+} else {
+  foreach ($path in $ScanPaths.Split(',')) {
```

Comment:
Why not capture the list of paths here as well and use that below?

Reply:
Done.